src/main/java/longah/node/Group.java

```diff
@@ -0,0 +1,122 @@
+package longah.node;
+
+import java.util.ArrayList;
+
+import longah.util.MemberList;
+import longah.util.Subtransaction;
+import longah.util.TransactionList;
+import longah.handler.StorageHandler;
+import longah.exception.LongAhException;
+import longah.exception.ExceptionMessage;
+
+public class Group {
+    private MemberList members;
+    private TransactionList transactions;
+    private StorageHandler storage;
+    private ArrayList<Subtransaction> transactionSolution;
+
+    /**
+     * Constructs a new Group instance with an empty member list and transaction list.
+     */
+    public Group() throws LongAhException {
+        this.members = new MemberList();
+        this.transactions = new TransactionList();
+        this.storage = new StorageHandler(this.members, this.transactions);
+    }
+
+    /**
+     * Sets the member list of the group.
+     * 
+     * @param members The member list to be set
+     */
+    public void setMemberList(MemberList members) {
+        this.members = members;
+    }
+
+    /**
+     * Returns the member list of the group.
+     * 
+     * @return The member list of the group
+     */
+    public MemberList getMemberList() {
+        return this.members;
+    }
+
+    /**
+     * Sets the transaction list of the group.
+     * 
+     * @param transactions The transaction list to be set
+     */
+    public void setTransactionList(TransactionList transactions) {
+        this.transactions = transactions;
+    }
+
+    /**
+     * Returns the transaction list of the group.
+     * 
+     * @return The transaction list of the group
+     */
+    public TransactionList getTransactionList() {
+        return this.transactions;
+    }
+
+    /**
+     * Update the transaction solution of the group based on the debts and credits of the members.
+     */
+    public void updateTransactionSolution() {
+        this.transactionSolution = this.members.solveTransactions();
+    }
+
+    /**
+     * Settles up the debts of the specified borrower by creating a transaction to repay all debts owed.
+     *
+     * @param borrowerName The name of the borrower to settle up.
+     */
+    public void settleUp(String borrowerName) throws LongAhException {
+        Member borrower = this.members.getMember(borrowerName);
+        if (borrower.getBalance() == 0) {
```

Comment:
what about the case where the person actually lent out money (ie balance of >0)? 

or if borrowerName is empty/doesnt exist in the member list

Reply:
Good sir >0 perhaps wont thrw exception and continue with the rest of the method...
Name not exist is dealt with by getmember